xskillscore/core/deterministic.py

```diff
@@ -683,14 +601,14 @@ def spearman_r_eff_p_value(a, b, dim=None, skipna=False, keep_attrs=False):
 
     Examples
     --------
-    >>> import numpy as np
-    >>> import xarray as xr
-    >>> from xskillscore import spearman_r_eff_p_value
-    >>> a = xr.DataArray(np.random.rand(5, 3, 3),
-                        dims=['time', 'x', 'y'])
-    >>> b = xr.DataArray(np.random.rand(5, 3, 3),
-                        dims=['time', 'x', 'y'])
-    >>> spearman_r_eff_p_value(a, b, dim='time')
+    >>> a = xr.DataArray(np.random.rand(5, 3, 3), dims=['time', 'x', 'y'])
+    >>> b = xr.DataArray(np.random.rand(5, 3, 3), dims=['time', 'x', 'y'])
+    >>> xs.spearman_r_eff_p_value(a, b, dim='time')
+    <xarray.DataArray (x: 3, y: 3)>
+    array([[0.4       ,        nan, 0.3       ],
+           [0.73802024, 0.7       , 0.7       ],
+           [0.80602663, 0.9       ,        nan]])
```

Comment:
Why are some Nan?

Reply:
Not sure. Never used `spearman_r_eff_p_value`. Is it a place holder for small values?

You probably want to `.isel(x=1, y=0)` to get the two vectors that go into that computation

Reply:
```
>>> _a = a.isel(x=1, y=0)
>>> _a
<xarray.DataArray (time: 5)>
array([0.59865848, 0.83244264, 0.13949386, 0.60754485, 0.44015249])

>>> _b = b.isel(x=1, y=0)
>>> _b
<xarray.DataArray (time: 5)>
array([0.54671028, 0.19598286, 0.14092422, 0.72900717, 0.31098232])
Dimensions without coordinates: time

_a2 = bn.nanrankdata(_a) # array([3., 5., 1., 4., 2.])
_b2 = bn.nanrankdata(_b) # array([4., 2., 1., 5., 3.])

rs = pearson_r(_a2, _b2) # 0.4
dof = 3
```